public/js/src/module/form-cache.js

```diff
@@ -30,6 +30,14 @@ let hash;
  * @return {Promise<Survey>}
  */
 function init( survey ) {
+    /**
+     * Note: it would probably be better if this were in enketo-webform.js, as it
+     * doesn't relate to the form cache per se. It's included here for testability.
+     */
+    if ( settings.type === 'preview' ) {
```

Comment:
This doesn't need to be `settings.type !== 'other'`?

Reply:
Bringing additional detail from Slack to consolidate discussion...

> I have one question inline. It’s not totally clear to me whether there’s any difference between preview and single, edit, view and only preview is tested.

My understanding from the [docs](https://apidocs.enketo.org/v2#/get-survey-all) was that `single` is a "single-submit" survey, which I just validated. However I now see that `single` is also "online-only", so I think you're right! In which case, I think the modification in connection.js is probably unnecessary, and both `formCache.init` and `getLastSavedRecord` could both perform this check to bypass except in `other` mode.